### PR TITLE
sql: fix ALTER PRIMARY KEY with virtual columns on the table

### PR DESCRIPTION
**Pull request by otan**
Release note (bug fix): Fix a bug where ALTER PRIMARY KEY would fail if
the table contained virtual columns.

### CODE REVIEW

**Review by rafiss — APPROVED** [review]
yay testing!! nice find

btw, optimizer team just added VIRTUAL columns this release, so i think only the 21.1 backport is needed.

File: pkg/sql/logictest/testdata/logic_test/alter_primary_key [hunk 1]
```diff
@@ -1404,3 +1404,24 @@ drop table child3;
...
+SHOW CREATE TABLE table_with_virtual_cols
+----
+table_with_virtual_cols  CREATE TABLE public.table_with_virtual_cols (
+                         id INT8 NOT NULL,
+                         new_pk INT8 NOT NULL,
+                         virtual_col INT8 NULL AS (1:::INT8) VIRTUAL,
+                         CONSTRAINT "primary" PRIMARY KEY (new_pk ASC),
+                         UNIQUE INDEX table_with_virtual_cols_id_key (id ASC),
+                         FAMILY fam_0_new_pk (new_pk),
+                         FAMILY fam_1_id (id)
```
[the remark this answers is not in the document]
Comment (otan) [reply pkg/sql/logictest/testdata/logic_test/alter_primary_key:1426]: hehe, this randomised family stuff man

**Comment by otan** [comment]
bors r=rafiss

**Comment by otan** [comment]
bors r=rafiss

**Comment by RaduBerinde** [comment]
Nice find, thanks for fixing!

**Pull request merged**